tests/engine/test_launch.py

```diff
@@ -141,6 +141,24 @@ def test_submit_store_provenance_false(self):
         with self.assertRaises(exceptions.InvalidOperation):
             launch.submit(AddWorkChain, term_a=self.term_a, term_b=self.term_b, metadata={'store_provenance': False})
 
+    def test_submit_process_function(self):
+        """Test the `submit` of process function."""
+        inputs = {'term_a': orm.Int(2), 'term_b': orm.Int(4)}
+        node = launch.submit(add, **inputs)
+        self.assertIsInstance(node, orm.CalcFunctionNode)
+
+        @calcfunction
+        def local_calcfunction(x: orm.Int):
+            return x + 1
+
+        res = launch.run(local_calcfunction, x=orm.Int(1))
+        self.assertEqual(res.value, 2)
+
+        # The local calcfunction can not be create, since class can
+        # not be found when create persistence from local declare function
+        # inputs = {'x': orm.Int(2)}
+        # node = launch.submit(local_calcfunction, **inputs)
```

Comment:
The `submit` can not apply to the locally announced calcfunction.

Reply:
This is to be expected though. The same holds exactly for all other processes, like `WorkChains`. If you define a `WorkChain` locally, either in a file that is not in the `PYTHONPATH` or let's say an interactive shell, when you submit it, it will fail, because the daemon cannot import the corresponding class.